cmd/vcluster/cmd/values/v1/base_values.go

```diff
@@ -0,0 +1,344 @@
+package values
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	policyv1 "k8s.io/api/policy/v1"
+)
+
+type BaseHelm struct {
+	GlobalAnnotations    map[string]interface{} `json:"globalAnnotations,omitempty"`
+	Pro                  bool                   `json:"pro,omitempty"`
+	EnableHA             bool                   `json:"enableHA,omitempty"`
+	Headless             bool                   `json:"headless,omitempty"`
+	DefaultImageRegistry string                 `json:"defaultImageRegistry,omitempty"`
+	Plugin               map[string]interface{} `json:"plugin,omitempty"`
```

Comment:
Would turning this into a `map[string]corev1.Container` make sense?

We could import the [corev1.Container definition](https://pkg.go.dev/k8s.io/api/core/v1#Container).

Reply:
I think the plugin definition is more than just the container. For eg. `rbac.role` etc.

Reply:
Can we create a new type embedding all the necessary fields and options in this case?

Reply:
I see, yeah that would be better. We can do that

Reply:
okay so an update on this, since the plugin name key can be a variable in this case, it would mean we can't just have a straight forward struct for this. We would need an overlapping struct OR need to implement a custom UnmarshalJSON function that extracts the name of plugin and then uses reflect to set that as the struct tag so that the plugin name se correctly set. I think we should just go with a map for this case and address it in future in case we really face customer demands about validation for plugin fields

Reply:
Why doesn't turning this into a `map[string]PluginSpec` work?

And `PluginSpec` could be something along the lines of


```go
type PluginSpec struct {
  corev1.Container
  rbacv1.Role
}
```

Reply:
Yes a map is the correct way here